### DISCUSSION TITLE
Installation on Shared Hosting

### DISCUSSION THREAD

**Discussion by emzee1** [Leantime/leantime#674] [Q&A] [answered]
Hi.

I am a small entrepreneur and intend to install Leantime on shared server (Solid CP from Mochahost).

I have followed all instructions given on https://docs.leantime.io/#/installation/quick-installation.

I have pointed my domain (aram.march7.group) to the public/ directory and its tested as http://aram.march7.group/images/bg1.png opens the image.

However i am unable to run http://aram.march7.group/install as it gives error 'The resource you are looking for has been removed, had its name changed, or is temporarily unavailable.'.

Please advise.

Thanking in anticipation.

Kind Regards

**Comment by marcelfolaron** [comment]
Do you see any errors in resources/logs? 
Can you try again with the latest version?

**Comment by pareshpandit** [comment]
I am facing a similar issue when trying to make things work with Hostinger's shared hosting, when tryying to do a quick install from the code download. I have created a subdomain. This is pointed to /pm folder in the main public_html folder for the domain, wherein, all LeanTime files have been added.

Have also installed a Zero SSL for the subdomain, successfully. So there is no issue wrt getting blocked for lack of it.
Have Cloudflare on the main domain. Tried pausing it, but no luck with that either. 

In Firefox, it just shows me a blank page, either @ https://pm.pareshpandit.com or trying to go for https://pm.pareshpandit.com/install . 
Only in an Opera or a Chrome will it give me an Error 500 : This domain is unable to handle this request.
I have also tried downgrading from PHP 7.4 to 7.3.

Interestingly, though, the LeanTime logo does show up as the favico on https://pm.pareshpandit.com, as you can see. 
So all is not lost. 

I am wondering what should now be the approach to getting around this issue.

I would also be grateful for to understand what exactly does pointing the domain to public/ directory mean?
Now, I have pointed my sub.domain.com to domain.com/sub (folder) in the public_html directory on the shared server for this domain. 
Does the above instruction mean the same as doing this, or is there something else that I am missing, I wonder?

BR/ Paresh

**Comment by marcelfolaron** [comment]
Hi,

First you would need to make sure that the subdomain points to the public
folder within Leantime. So in your case

public_html/pm/public

Next please check if you see any errors in resources/logs/error.log

If you don’t see any errors in there please check the error logs of your
hosting provider

One other observation: I just helped someone with a similar set up and due
to some hosting specific redirects adding Leantime into public_html caused
some issues. After I moved the directory out (to the root next to
public_html) it started working. Maybe worth a try.

**Reply by pareshpandit** [reply]
Hi Marcel,

Many thanks for the quick and helpful reply. 
I shall try to incorporate these changes, and report back.

At the moment, turning on errors showed that the 500 Internal Error was occurring due to the installation not finding the 
'../vendor/autoload.php' file, called from Line 43 of our 'src/core/class.autoload.php'.
Not only is this path non-existent, but the file (autoload.php) is not present in the entire package.
So commenting it off (for testing), at least produced the installation page. @ sub.domain.com/install 

From there on, the installation seemed to do nothing.
From your response, I am hoping that pointing to the public folder might help.
(Or maybe that autoload.php needs to appear from somewhere ?)

I am new to doing this, and appreciate all the help and guidance.
While I re-point the sub.domain.com, would appreciate if this rings a bell to someone, and if they can guide in the meanwhile.

[Also found that this particular thing -- missing ../vendor/autoload.php -- happens with Laravel, too. Is this a build error of some sort?]

BR/ **Paresh**

**Comment by marcelfolaron** [comment]
What package did you use to install Leantime? Did you use the official
download package?

Either here: https://github.com/Leantime/leantime/releases/tag/v2.1.9
Or here: https://leantime.io/download/

You can't use the "source" package without running a few commands to build
the entire site (eg running composer, npm and grunt tasks). This would be
considered the development version.

**Reply by pareshpandit** [reply]
Yes, I downloaded from the Github Rep.

Also, update: 
I pointed the sub.domain.com to the /public folder.
The missing file reference is still commented out.

I also added SMTP settings to the config.php.

The DB got created, as I can check from myPHP Admin.

However, seems like the install registration is still faliling.
I am not getting a confirmation or approval email, and neither is the install option disappearing, or letting me login with the submitted credentials.

**Reply by ashelman-ec** [reply]
> https://leantime.io/download/

@marcelfolaron I has also having problems installing, and this comment from you (to not use the 'source' package but to take the release package from the download page) solved my problem. I had taken the git repository tagged v2.1.9 directly.

I suggest that you also highlight this in your installation instructions at https://docs.leantime.io/#/installation/quick-installation

Perhaps you could add the link to the download page in the first item: 1. [Download](https://leantime.io/download/) latest release package

All: The symptom of this mistake is getting a blank page. If that's what you have, then try this to solve it.

**Reply by gloriafolaron** [reply]
@ashelman-ec thanks for the call out on this.  I’m going to circle back with @marcelfolaron on looking at adding that clarity to the docs.

**Comment by marcelfolaron** [comment]
And you say the "Vendor" folder is missing?
If so please download/upload the package again (The one called
"leantime-v2.1.9.zip") Something must have gone wrong with the upload.

**Comment by pareshpandit** [comment]
It was my bad. After you pointed out, I checked and found that I had installed the Master instead of the release.
So I replaced it with the v2.1.9 download from page that was the github link shared by you.

Now, interestingly enough, the error is that the file referencing that file in Vendor, is missing. 

**It now reads:** 
Warning: include_once(../src/core/class.autoload.php): failed to open stream: No such file or directory in /.../.../.../public/index.php on line 8
Warning: include_once(): Failed opening '../src/core/class.autoload.php' for inclusion (include_path='.:/opt/alt/php73/usr/share/pear') in /.../.../.../public/index.php on line 8

Line 8 @ index.php reads : include_once '../src/core/class.autoload.php';
[ And I checked, the class.autoload.php is indeed missing --- which was calling the other autoload php file from the Vendor directory, in the earlier (Master) installation. ]

But this time, if I comment this line out, I am left with a blank page (sans Errors), and only the favicon in the title.

:-/

**Comment by pareshpandit** [comment, marked as answer]
Let me try to reinstall in some other fashion, and get back to you in a bit. Maybe the problem is with uploads, as I can see the class.autoload.php in local extraction of the zip.

Apologies.

**Comment by pareshpandit** [comment]
**UPDATE :** 

All looks good and healthy right now. :) 

Revised the DB too.
After Admin Setup, the '.../install' also doesn't show up. So that's not broken either.
The login worked, and Dashboard showed up. 

Looking forward to trying it out.

Thank you so much, for the guidance, and even more importantly, the kind patience! 

BR/ **Paresh**

**Comment by FrankDavidTaylor** [comment]
I also faced too many issues while adding domain names to my Laravel application and it took too much time from my calendar. I recently found the Devrims [Managed Laravel Hosting](https://devrims.com/laravel-hosting/) platform, where you can install the Laravel application in a few clicks and add multiple domain names. Try Devrims via creating Free Trial Account.

**Reply by marcelfolaron** [reply]
You do realize this is not a Laravel application? Don't appreciate the spam here.

**Comment by marcelfolaron** [comment]
can you check if src/core/class.autoload.php is there? It should be in the
local package that you downloaded. If it is missing, please upload again.
Commenting out the line won't help because that file is required to run the
system.
